src/fuzzing_main.cc

```diff
@@ -0,0 +1,130 @@
+#include "src/monitors/gdb_monitor.h"
+
+static bool ValidatePort(const char* flagname, gflags::int32 value) {
+  return value > 0 && value < 65535;
+}
+
+static bool ValidateMonitor(const char* flagname, const std::string& value) {
+  const absl::flat_hash_set<std::string> kSupportedMonitors = {
+      "blackbox", "cortexm4_gdb", "gdb"};
+  return kSupportedMonitors.contains(value);
+}
+
+static bool ValidateFuzzingMode(const char* flagname,
+                                const std::string& value) {
+  const absl::flat_hash_set<std::string> kSupportedModes = {
+      "cbor_make_credential", "cbor_get_assertion", "cbor_client_pin",
+      "ctap2_raw"};
+  return kSupportedModes.contains(value);
+}
+
+DEFINE_string(
+    token_path, "",
+    "The path to the device on your operating system, usually /dev/hidraw*.");
+
+DEFINE_string(
+    corpus_path, "corpus_tests/test_corpus/",
+    "The path to the corpus containing seed files to test the device.");
+
+DEFINE_string(monitor, "blackbox", "The monitor type used in fuzzing.");
+DEFINE_string(fuzzing_mode, "cbor_make_credential",
+              "The type of inputs to be fuzzed.");
+
+DEFINE_bool(verbose, false, "Printing debug logs, i.e. transmitted packets.");
+
+DEFINE_int32(port, 2331, "Port to listen on for GDB remote connection.");
+DEFINE_int32(num_runs, 0,
+             "Number of inputs to be run. By default, the fuzzer will run "
+             "indefinitely.");
+DEFINE_int32(max_length, 0,
+             "Maximum length of an input. By default, there is no limit.");
+DEFINE_int32(max_mutation_degree, 10,
+             "Maximum number of successive mutation operations to be applied. "
+             "By default, value is 10.");
+DEFINE_int32(seed, time(NULL), "The random seed for all random operations.");
+
+DEFINE_validator(port, &ValidatePort);
+DEFINE_validator(monitor, &ValidateMonitor);
+DEFINE_validator(fuzzing_mode, &ValidateFuzzingMode);
+
+int main(int argc, char** argv) {
+  google::InitGoogleLogging(argv[0]);
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+  if (FLAGS_token_path.empty()) {
+    std::cout << "Please add the --token_path flag for one of these devices:"
+              << std::endl;
+    fido2_tests::hid::PrintFidoDevices();
+    return 0;
+  }
+  if (FLAGS_token_path == "_") {
+    // This magic value is used by the run script for comfort.
+    FLAGS_token_path = fido2_tests::hid::FindFirstFidoDevicePath();
+    std::cout << "Testing device at path: " << FLAGS_token_path << std::endl;
+  }
+
+  fido2_tests::DeviceTracker tracker;
+  std::unique_ptr<fido2_tests::DeviceInterface> device =
+      std::make_unique<fido2_tests::hid::HidDevice>(&tracker, FLAGS_token_path,
+                                                    FLAGS_verbose);
+  CHECK(fido2_tests::Status::kErrNone == device->Init())
+      << "CTAPHID initialization failed";
+  device->Wink();
+  std::cout << "This tool will irreversibly delete all credentials on your "
```

Comment:
Good point. I'm mentioning @kaczmarczyck for the code owner's decision.

Reply:
The `Reset` command always needs a user input. So as long as you don't accept that, it's fine. Red is usually used to indicate failed tests, so maybe we can have some other bright color like `\u001b[35m`? Please change all other main implementations accordingly, if you want to go that route. Perhaps it would be best to integrate it into the command state initialization, since this is where the `Reset` occurs. And then you get the warning for all `main`s for free.

Reply:
I can do that separately if you want, since it involves other modules

Reply:
I'm okay with postponing. @jennypullman ?